src/main/kotlin/org/opensearch/commons/notifications/model/SNS.kt

```diff
@@ -0,0 +1,114 @@
+/*
+ * SPDX-License-Identifier: Apache-2.0
+ *
+ * The OpenSearch Contributors require contributions made to
+ * this file be licensed under the Apache-2.0 license or a
+ * compatible open source license.
+ *
+ * Modifications Copyright OpenSearch Contributors. See
+ * GitHub history for details.
+ */
+package org.opensearch.commons.notifications.model
+
+import org.opensearch.common.io.stream.StreamInput
+import org.opensearch.common.io.stream.StreamOutput
+import org.opensearch.common.io.stream.Writeable
+import org.opensearch.common.xcontent.ToXContent
+import org.opensearch.common.xcontent.XContentBuilder
+import org.opensearch.common.xcontent.XContentParser
+import org.opensearch.common.xcontent.XContentParserUtils
+import org.opensearch.commons.utils.fieldIfNotNull
+import org.opensearch.commons.utils.logger
+import java.io.IOException
+import java.util.regex.Pattern
+
+/**
+ * SNS notification data model
+ */
+data class SNS(val topicARN: String, val roleARN: String?) : BaseConfigData {
+
+    init {
+        require(SNS_ARN_REGEX.matcher(topicARN).find()) { "Invalid AWS SNS topic ARN: $topicARN" }
+        if (roleARN != null) {
+            require(IAM_ARN_REGEX.matcher(roleARN).find()) { "Invalid AWS role ARN: $roleARN " }
+        }
+    }
+
+    override fun toXContent(builder: XContentBuilder, params: ToXContent.Params): XContentBuilder {
+        return builder.startObject()
+            .field(TOPIC_ARN_FIELD, topicARN)
+            .fieldIfNotNull(ROLE_ARN_FIELD, roleARN)
+            .endObject()
+    }
+
+    /**
+     * Constructor used in transport action communication.
+     * @param input StreamInput stream to deserialize data from.
+     */
+    constructor(input: StreamInput) : this(
+        topicARN = input.readString(),
+        roleARN = input.readOptionalString()
+    )
+
+    @Throws(IOException::class)
+    override fun writeTo(out: StreamOutput) {
+        out.writeString(topicARN)
+        out.writeOptionalString(roleARN)
+    }
+
+    companion object {
+        private val log by logger(SNS::class.java)
+
+        private val SNS_ARN_REGEX =
+            Pattern.compile("^arn:aws(-[^:]+)?:sns:([a-zA-Z0-9-]+):([0-9]{12}):([a-zA-Z0-9-_]+)$")
+        private val IAM_ARN_REGEX = Pattern.compile("^arn:aws(-[^:]+)?:iam::([0-9]{12}):([a-zA-Z_0-9+=,.@\\-_/]+)$")
+
+        const val TOPIC_ARN_FIELD = "topic_arn"
+        const val ROLE_ARN_FIELD = "role_arn"
+        const val SNS_TYPE = "sns"
```

Comment:
I think we put constants in `NotificationConstants.kt` and import elsewhere.

Reply:
updated